File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        LinkedList<Integer> queue = new LinkedList<>();
        StringBuilder sb = new StringBuilder();
        while(n-- > 0){
            StringTokenizer st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken();

            if(cmd.equals("push")){
                int num = Integer.parseInt(st.nextToken());
                queue.offer(num);
            } else if (cmd.equals("pop")){
                if(queue.isEmpty()) sb.append(-1).append("\n");
                else sb.append(queue.poll()).append("\n");
            } else if (cmd.equals("size")) {
                sb.append(queue.size()).append("\n");
            } else if (cmd.equals("empty")){
                if(queue.isEmpty()) sb.append(1).append("\n");
                else sb.append(0).append("\n");
            } else if (cmd.equals("front")){
                if(queue.isEmpty()) sb.append(-1).append("\n");
                else sb.append(queue.getFirst()).append("\n");
            } else if (cmd.equals("back")){
                if(queue.isEmpty()) sb.append(-1).append("\n");
                else sb.append(queue.getLast()).append("\n");
            }
        }
        
        System.out.println(sb);
    }
    
}

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <bits/stdc++.h>
using namespace std;

int main(void){
  ios::sync_with_stdio(0);
  cin.tie(0);
  
  int n;
  cin >> n;

  queue<int> q;
  for(int i=0; i<n; i++){
    string cmd;
    cin >> cmd;

    if(cmd == "push"){
      int num; cin >> num;
      q.push(num);
    } else if (cmd == "pop"){
      if(q.empty()) cout << -1 << '\n';
      else {
        cout << q.front() << '\n';
        q.pop();
      }
    }
    else if (cmd == "front") q.empty() ? cout << -1 << '\n' : cout << q.front() << '\n';
    else if (cmd == "back") q.empty() ? cout << -1 << '\n' : cout << q.back() << '\n';
    else if (cmd == "size") cout << q.size() << '\n';
    else if (cmd == "empty") q.empty() ? cout << 1 << '\n' : cout << 0 << '\n';
  }

  return 0;
}